### PR TITLE
Comment Storage Update in APIView

### PR DESCRIPTION
**Pull request by mccoyp**
This change switches from preview 1 of Azure.Storage.Blobs to preview 2, and changes how comments are stored in blobs. Rather than storing each comment as a standalone blob, comments for each review are grouped into a single blob.

### CODE REVIEW

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml.cs [hunk 1]
```diff
@@ -54,24 +54,32 @@ public async Task OnGetAsync(string id)
             var assemblyModel = await assemblyRepository.ReadAssemblyContentAsync(id);
             var renderer = new HTMLRendererAPIV();
             AssemblyModel = renderer.Render(assemblyModel.Assembly).ToArray();
-            var comments = await commentRepository.FetchCommentsAsync(id);
-
             Comments = new Dictionary<string, List<CommentModel>>();
-            foreach (var comment in comments)
+
+            var assemblyComments = await commentRepository.FetchCommentsAsync(id);
+            if (assemblyComments != null)
```
Comment (pakrym) [inline src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml.cs]: Why would it be null?

File: src/dotnet/APIView/APIViewWeb/BlobCommentRepository.cs [hunk 2]
```diff
@@ -23,53 +23,129 @@ public BlobCommentRepository(IConfiguration configuration)
...
+        /// </summary>
+        /// <param name="id">The ID of the blob to have its comments read.</param>
+        /// <returns>The comments existing in the specified blob.</returns>
+        public async Task<AssemblyCommentsModel> ReadBlobContentAsync(string id)
         {
-            var guid = Guid.NewGuid().ToString();
-            commentModel.Id = guid;
-            var blob = ContainerClient.GetBlockBlobClient(guid);
+            var result = await ContainerClient.GetBlobClient(id).DownloadAsync();
+            //result.Value.Properties.ETag   Can use this to check if version of assembly comments is up to date
```
Comment (pakrym) [inline src/dotnet/APIView/APIViewWeb/BlobCommentRepository.cs]: Rem

File: src/dotnet/APIView/APIViewWeb/BlobCommentRepository.cs [hunk 3]
```diff
@@ -23,53 +23,129 @@ public BlobCommentRepository(IConfiguration configuration)
...
+        /// <param name="assemblyId">The ID of the assembly to have its comments read.</param>
+        /// <returns>The comments existing for the specified assembly if it exists, or null if no assembly has the specified ID.</returns>
+        public async Task<AssemblyCommentsModel> FetchCommentsAsync(string assemblyId)
         {
-            var segment = await ContainerClient.ListBlobsFlatSegmentAsync(options: new BlobsSegmentOptions() { Details = new BlobListingDetails() { Metadata = true } });
+            var blobs = await FetchBlobsAsync();
 
-            var comments = new List<CommentModel>();
-            foreach (var item in segment.Value.BlobItems.OrderBy(blob => blob.Properties.CreationTime))
+            foreach (var blob in blobs.OrderBy(blob => blob.Properties.CreationTime))
```
Comment (pakrym) [inline src/dotnet/APIView/APIViewWeb/BlobCommentRepository.cs]: You are still doing an inefficient thing of enumerating all blobs to find the right one.

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml.cs (hunk 1 above)
Comment (mccoyp) [reply src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml.cs]: Currently, a blob isn't created to store review comments until a comment is uploaded, so `FetchCommentsAsync()` will return null the first time a review is opened.

File: src/dotnet/APIView/APIViewWeb/BlobCommentRepository.cs (hunk 3 above)
Comment (pakrym) [reply src/dotnet/APIView/APIViewWeb/BlobCommentRepository.cs]: Yeah, reuse the GUID

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml.cs (hunk 1 above)
Comment (pakrym) [reply src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml.cs]: Return an empty object then.

File: src/dotnet/APIView/APIViewWeb/BlobCommentRepository.cs [hunk 4]
```diff
@@ -106,10 +62,9 @@ public async Task UploadCommentAsync(CommentModel commentModel, string assemblyI
             commentModel.Id = Guid.NewGuid().ToString();
 
             AssemblyCommentsModel assemblyComments = await FetchCommentsAsync(assemblyId);
-            if (assemblyComments == null)
+            if (assemblyComments.AssemblyId == null)
```
Comment (pakrym) [inline src/dotnet/APIView/APIViewWeb/BlobCommentRepository.cs]: This if is strange.

File: src/dotnet/APIView/APIViewWeb/Models/AssemblyCommentsModel.cs [hunk 5]
```diff
@@ -6,15 +6,19 @@ public class AssemblyCommentsModel
...
         public AssemblyCommentsModel() { }
 
-        public AssemblyCommentsModel(string assemblyId, CommentModel commentModel, string id)
+        public AssemblyCommentsModel(string assemblyId)
+        {
+            this.AssemblyId = assemblyId;
+            this.Comments = new CommentModel[] { };
+        }
+
+        public AssemblyCommentsModel(string assemblyId, CommentModel commentModel)
```
Comment (pakrym) [inline src/dotnet/APIView/APIViewWeb/Models/AssemblyCommentsModel.cs]: This ctor is strange. Just create an instance and call AddComment on it.

**Pull request merged**